02-microarray/gene-id-annotation_microarray_01_ensembl.Rmd

```diff
@@ -0,0 +1,368 @@
+mapped_df <- mapped_list %>%
+  reshape2::melt()
+```
+
+Now let's take a peek at our data frame.
+
+```{r}
+head(mapped_df)
+```
+
+We can see that our data frame has a new column `value`.
+Let's get a summary of the gene symbols returned in the `value` column of our mapped data frame.
+
+```{r}
+# We can use the `summary()` function to get a better idea of the distribution of symbols in the `value` column
+summary(mapped_df$value)
+```
+
+There are 998 NA's in our data frame, which means that 998 out of the 17918 Ensembl IDs did not map to gene symbols.
+998 out of 17918 is not too bad a rate, in our opinion, but note that different gene identifier types will have different mapping rates and that is to be expected.
+Regardless, it is always good to be aware of how many genes you are potentially "losing" if you rely on this new gene identifier you've mapped to for downstream analyses.
+
+However, if you have almost all NA's it is possible that the function was executed incorrectly or you may want to consider using a different gene identifier, if possible.
+
+Now let's check to see if we have any genes that were mapped to multiple symbols.
+
+```{r}
+multi_mapped <- mapped_df %>%
+  # Let's group by the Ensembl ID's in the `L1` column
+  dplyr::group_by(L1) %>%
+  # Create a new variable containing the number of symbols mapped to each ID
+  dplyr::mutate(gene_symbol_count = dplyr::n()) %>%
+  # Arrange by the genes with the highest number of symbols mapped
+  dplyr::arrange(desc(gene_symbol_count)) %>%
+  # Filter to include only the rows with multi mappings
+  dplyr::filter(gene_symbol_count > 1)
+
+# Let's look at the first 6 rows of our `multi_mapped` object
+head(multi_mapped)
+```
+
+Looks like we have some cases where 3 gene symbols mapped to a single Ensembl ID.
+We have a total of 130 out of 17984 Ensembl IDs with multiple mappings to gene symbols.
+We are not too worried about the 130 IDs with multiple mappings, so we will filter them out for the purpose of having 1:1 mappings for our downstream analysis.
+
+## Map Ensembl IDs to gene symbols -- filtering out multi mappings
+
+The next code chunk will rerun the `mapIds()` function, this time supplying the `"filter"` option to the `multiVals` argument.
+This will remove all instances of multiple mappings and return a list of only the gene identifiers and symbols that had 1:1 mapping.
+Use `?mapIds` to see more options or strategies.
+
+```{r}
+# Map ensembl IDs to their associated gene symbols
+filtered_mapped_df <- mapIds(
+  org.Mm.eg.db, # Replace with annotation package for the organism relevant to your data
+  keys = df$Gene,
+  column = "SYMBOL", # Replace with the type of gene identifiers you would like to map to
+  keytype = "ENSEMBL", # Replace with the type of gene identifiers in your data
+  multiVals = "filter"
+) %>%
+  # Turn our list into a data frame
+  reshape2::melt() %>%
```

Comment:
I might be wrong, but I don't think this melt is necessary, I *think* you should be able to put this straight into a data frame since it should be the same length as the original data.